pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -18,27 +18,22 @@ const (
 	ImageLabel       = "image.build.pivotal.io/image"
 
 	BuildReasonAnnotation = "image.build.pivotal.io/reason"
-	BuildReasonConfig     = "CONFIG"
-	BuildReasonCommit     = "COMMIT"
-	BuildReasonBuildpack  = "BUILDPACK"
-	BuildReasonStack      = "STACK"
-)
-
-func (im *Image) buildNeeded(lastBuild *Build, sourceResolver *SourceResolver, builder BuilderResource) ([]string, bool, error) {
-	if !sourceResolver.Ready() {
-		return []string{}, false, nil
-	}
+	BuildNeededAnnotation = "image.build.pivotal.io/additionalBuildNeeded"
 
-	if !builder.Ready() {
-		return []string{}, false, nil
-	}
+	BuildReasonConfig    = "CONFIG"
+	BuildReasonCommit    = "COMMIT"
+	BuildReasonBuildpack = "BUILDPACK"
+	BuildReasonStack     = "STACK"
+	BuildReasonTrigger   = "TRIGGER"
+)
 
-	if lastBuild == nil {
-		return []string{BuildReasonConfig}, true, nil
+func (im *Image) buildNeeded(lastBuild *Build, sourceResolver *SourceResolver, builder BuilderResource) ([]string, bool) {
+	if !sourceResolver.Ready() || !builder.Ready() {
+		return []string{}, false
 	}
 
-	if im.Spec.Tag != lastBuild.Tag() {
-		return []string{BuildReasonConfig}, true, nil
+	if lastBuild == nil || im.Spec.Tag != lastBuild.Tag() {
```

Comment:
Given `im.Spec.Tag` is immutable, when would it not be equal to the `lastBuild.Tag()`? Is this assuming that maybe we don't have a validating webhook running?

Reply:
Good Question. I left that functionality in here because, it seems possible for the tag to change perhaps when the webhook is not running. Open to removing it though.